docs/developer/solutions/archive.rst

```diff
@@ -0,0 +1,152 @@
+Solution archive guidelines
+===========================
+
+To provide a predictable interface with packaged Solutions, MetalK8s expects a
+few criteria to be respected, described below.
+
+
+Archive format
+--------------
+
+Solution archives must use the `ISO-9660:1988`_ format, including `Rock Ridge`_
+and Joliet_ directory records. The character encoding must be UTF-8_. The
+conformance level is expected to be at most 3, meaning:
+
+- Directory identifiers may not exceed 31 characters in length
```

Comment:
What's a characters here: A byte (I suppose)? A Unicode codepoint? A grapheme cluster (highly unlikely)?

Since we're allowing/requiring UTF-8 here, a character is something ambiguous and may not match what people expect.

Reply:
Nothing clear from what I read so far, feel free to go through the [specs](http://www.ecma-international.org/publications/standards/Ecma-119.htm) ;). In the meantime, I'll mention `bytes`.

Reply:
TBH, I just read through the docs and tried to translate the command we use in the buildchain into some text, but I have no clue if what I wrote is correct (we probably don't care much if we can mount and read from the ISO).

Reply:
IIUC, ISO 9660 doesn't support Unicode characters so there is no issue (one character == one byte, hence the restrictions expressed in term of character).
Joliet use UCS-2 encoding (which doesn't support all the Unicode characters).

I guess that the UTF-8 is supported through Rock Ridge extensions.

Seems complicated, so let's use mention `bytes` as you suggest :D